text/0000-innerhtml-binding.md

```diff
@@ -78,17 +85,41 @@ some content defined, like bellow:
 
 ```html
 <template>
-    <div lwc:innerHTML={myHTML()}>
+    <div lwc:inner-html={myHTML()}>
       Adding content when innerMTML is defined leads to a compilation error
     </div>
 </template>
 ```
 
+### Validity
+
+The `lwc:inner-html` attribute can be set on any tag in an LWC template with the
+following restrictions:
+
+  - The host element cannot contain both declarative markup and the `lwc:inner-html`
+  attribute.  
+  - The `lwc:inner-html` attribute cannot be applied to LWC components, as their
+  content is defined by a template.  
+  - The `lwc:inner-html` attribute cannot be applied to the <slot> tag, as its content
+  comes from a slot.  
+
+
+### Implementation details
+
+From an implementation standpoint, the HTML fragment is kept in the host element
+attribute. When the VDOM comparison finds that this attribute has changed, then 
+it updates the DOM element content by assigning it to the element.innerHTML property.
+
 
 ## Drawbacks
 
-- It might open security breaches if not used appropriately. The proposed API forces the developer to know what he/she is doing and avoids mistaken assignment.  
-- Locker might prevent this assignment. To be verified.
+  - It might open security breaches if not used appropriately. The proposed API forces 
+  the developer to know what he/she is doing and avoids mistaken assignment.  
+  - Locker might prevent this assignment. To be verified.  
+
+Note: the security issue is *not* different from setting the HTML with a DOM operation

```

Comment:
That's correct, but the difference is that the user is setting it vs the framework is setting it. Locker only controls the user code, framework on the other hand is system mode, and not controlled by locker.

This touches on another RFC for LWC that is pending to be written, which is set to provide hooks that libraries like locker can take control over to sanitize injections/sinks. That's TBD.

Reply:
Ok we need more info then. Is there a way to detect that locker is available and then call its sanitization methods? Sounds like hard-coding it but might be a workaround before a more generic solution.